### DISCUSSION TITLE
Manifest mode and update of the vcpkg-configuration.json

### DISCUSSION THREAD

**Discussion by pavel-pokutnev** [microsoft/vcpkg#30011] [Q&A]
I'm looking for help regarding automatic port installation in case the `vcpkg-configuration.json` is getting updated.

Please consider the following scenario:
- Visual Studio 2022 17.5.1 C++ project with vcpkg manifest enabled.
- `vcpkg.json` is present.
- `vcpkg-configuration.json` is present (for custom registry).
- On new checkout, the installation happening completely automatic and `.\vcpkg_installed` is created.
- Everything is working as expected!

The time goes by, and a new port is added into the custom registry.

- `vcpkg.json` is getting updated with new port entries.
- `vcpkg-configuration.json` is getting updated with the new baseline and packages.
- **Expectation**: installation of the new dependency shall happen automatically during the next compile or open of the solution. 
- **Issue**: The expected installation is failing with the following error:

``
1>Couldn't find the versions database file: C:\Users\<User>\AppData\Local\vcpkg\registries\git-trees\bee7f6e5461c66d843f71f2n457z86223a6dm0bd\n-\new-port.json
``

It looks like the vcpkg looks into the `.\vcpkg_installed\x64-windows\vcpkg\vcpkg-lock.json` for the registry git-tree (with the older version) instead to use the new baseline from `vcpkg-configuration.json`.

As a workaround, you can delete the `.\vcpkg_installed` and the vcpkg will regenerate it on the next install.

My question is, is it possible to overcome this issue somehow, so the installation will happen automatically?

Is there any documentation for the `vcpkg-lock.json`, or can someone shortly explain the need for this file?

**Comment by j-bruel** [comment]
Do you have any update? We facing the same issue.

**Reply by pavel-pokutnev** [reply]
@j-bruel Hi! Not yet. Have you found any workaround?

**Comment by stormy9898** [comment]
Anyone figure it out?

**Comment by luncliff** [comment]
I think the message is requesting `versions/` folder to be updated.
Would you check if the registry updated the folder together?

An update commit after this command may solve the problem
```ps1
vcpkg x-add-version $PortName `
        --vcpkg-root="$VcpkgRoot" `
        --x-builtin-ports-root="$PortRoot" `
        --x-builtin-registry-versions-dir="$VersionRoot"
```
